tests/Makefile.tests_common

```diff
@@ -1,7 +1,7 @@
-ifneq (,$(filter driver_%,$(APPLICATION)))
+APPLICATION ?= tests_$(notdir $(patsubst %/,%,$(CURDIR)))
```

Comment:
Should we be stricter here and **not** allow another `APPLICATION` definition in the tests? IMO this ensures consistency .. `make` should fail if you remove the `?` and a test application defines its own `APPLICATION`, right?

Reply:
A good idea for the next step! ;)

There are a some tests (openthread, lwmac) which do not even use tests/Makefile.tests_common. We can unify them (as followup), then force as you suggested. But not in this PR.

Reply:
>A good idea for the next step! ;)

I'll take your word for that (: